### PR TITLE
DateTime, Guid and Task Delay roslyn code fixers

### PR DESCRIPTION
**Pull request by allantargino**
Following the same structure as #318, this PR adds roslyn code fixers for the diagnostics reported about non-deterministic usage of DateTime, Guid and Task delay. It suggests replacing them for the `TaskOrchestrationContext` alternatives, as suggested by [our docs](https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-code-constraints?tabs=csharp).

### CODE REVIEW

File: test/Analyzers.Tests/Orchestration/DateTimeOrchestrationAnalyzerTests.cs [hunk 1]
```diff
@@ -45,7 +45,6 @@ public async Task NonOrchestrationHasNoDiag()
     [Theory]
     [InlineData("DateTime.Now")]
     [InlineData("DateTime.UtcNow")]
-    [InlineData("DateTime.Today")]
```
Comment (bachuv) [inline test/Analyzers.Tests/Orchestration/DateTimeOrchestrationAnalyzerTests.cs:48]: Why did you remove this line?

Comment (allantargino) [reply test/Analyzers.Tests/Orchestration/DateTimeOrchestrationAnalyzerTests.cs:48]: I extract that use case into a brand new `Fact` unit test. I did it because the expected fix for `DateTime.Today` is different from `Now` and `UtcNow`. So it was more a refactor than removing a test.

**Review by bachuv — APPROVED** [review]
LGTM!
